base_layer/core/src/validation/block_validators.rs

```diff
@@ -125,6 +125,19 @@ impl<B: BlockchainBackend> PostOrphanBodyValidation<B> for BodyOnlyValidator {
     /// 1. Are all inputs and outputs not in the STXO set?
     /// 1. Are the block header MMR roots valid?
     fn validate_body_for_valid_orphan(&self, block: &ChainBlock, backend: &B) -> Result<(), ValidationError> {
+        let metadata = backend.fetch_chain_metadata()?;
```

Comment:
I don't think this is the correct place to do this. This validator would prevent any kind of reorg if this is the case. I would say move this code to the GetBlockTemplate in the inbound handler

Reply:
Changes discussed via DM